util/testutil/addcat/addcat.go

```diff
@@ -0,0 +1,39 @@
+package testutil
+
+import (
+	"bytes"
+	"strings"
+
+	core "github.com/ipfs/go-ipfs/core"
+	coreunix "github.com/ipfs/go-ipfs/core/coreunix"
+	unixfs "github.com/ipfs/go-ipfs/shell/unixfs"
+	errors "github.com/ipfs/go-ipfs/util/debugerror"
+)
+
+func AddCat(adder *core.IpfsNode, catter *core.IpfsNode, data string) error {
```

Comment:
On Sat, May 02, 2015 at 02:43:28AM -0700, Jeromy Johnson wrote:
> > +func AddCat(adder *core.IpfsNode, catter *core.IpfsNode, data string) error {
> 
> I think we will want to use a `[]byte` here simply because so many
> other potential callers of this will generate data in `[]byte` form,
> and turning that into a `string` type requires a full copy of the
> entire data. For larger `AddCat` tests, this may be copying hundreds
> of MBs of data, not something i'm a fan of

This cuts the other way too if our potential callers are generating
strings and need to convert them to []bytes.  Why are the majority of
our callers generating mutable data?